crates/rooch-rpc-server/src/server/rooch_server.rs

```diff
@@ -642,19 +645,49 @@ impl RoochAPIServer for RoochServer {
             .map(|m| m.object_id.clone())
             .collect::<Vec<_>>();
         let access_path = AccessPath::objects(object_ids.clone());
-        let mut data = self
-            .rpc_service
-            .get_annotated_states(access_path)
-            .await?
+        let annotated_states = self.rpc_service.get_annotated_states(access_path).await?;
```

Comment:
If the `ObjectStateFilterView` is `ObjectStateFilterView::ObjectID,` we should directly build the `AccessPath,` and do not need to call the `query_object_states.`

Reply:
indeed

Reply:
The Object returned from the states API does not have a timestamp. The timestamp can be returned from the Transaction API. Consider adding a timestamp to `Transaction`?